monitor/seaclouds-data-collector/src/main/java/eu/seaclouds/monitor/seacloudsDc/Metric.java

```diff
@@ -93,29 +78,61 @@ public void update(Observable o, Object arg) {
         }
     }
 
+    private String getApplicationUser(Resource resource){
+        try {
+            if(getParameters(resource).get(APPLICATION_USER_PARAMETER) != null){
+                return getParameters(resource).get(APPLICATION_USER_PARAMETER);
+            }else{
+                return new String();
+            }
+        } catch (Exception e) {
+            logger.error("Error while reading the applicationUser", e);
+            throw new IllegalStateException("Error while reading the applicationUser",e.getCause());
+        }    
+    }
+    
+    private String getApplicationPassword(Resource resource){
+        try {
+            if(getParameters(resource).get(APPLICATION_PASSWORD_PARAMETER) != null){
+                return getParameters(resource).get(APPLICATION_PASSWORD_PARAMETER);
+            }else{
+                return new String();
+            }        
+        } catch (Exception e) {
+            logger.error("Error while reading the applicationPassword", e);
+            throw new IllegalStateException("Error while reading the applicationPassword",e.getCause());
+        }    
+    }
+    
     private int getSamplingTime(Resource resource) {
         if (getParameters(resource) == null
-                || getParameters(resource).get("samplingTime") == null)
+                || getParameters(resource).get(SAMPLING_TIME_PARAMETER) == null)
             return DEFAULT_SAMPLING_TIME;
         try {
             return Integer
-                    .parseInt(getParameters(resource).get("samplingTime"));
+                    .parseInt(getParameters(resource).get(SAMPLING_TIME_PARAMETER));
         } catch (Exception e) {
             logger.error("Error while reading the sampling time", e);
             return DEFAULT_SAMPLING_TIME;
         }
     }
 
-    private void createTask(Timer timer, Resource resource, int samplingTime) {
-        timer.scheduleAtFixedRate(new MetricSender(resource), 0,
+    private void createTask(Timer timer, Resource resource, int samplingTime, String applicationUrl, String user, String password) {
+        timer.scheduleAtFixedRate(new MetricSender(resource, applicationUrl, user, password), 0,
                 samplingTime * 1000);
     }
 
     private class MetricSender extends TimerTask {
```

Comment:
I'm still not completely convinced about the motivation behind this refactoring. My assumption was to have that class private within the abstract class because I don't want it to be available outside. Probably I'm wrong and I don't see your point, but I would not change this, at least without envisioning specific pros.